apps/meteor/client/views/room/MessageList/hooks/useMessageBody.tsx

```diff
@@ -1,14 +1,39 @@
-import type { IMessage } from '@rocket.chat/core-typings';
+import type { IMessage, ISubscription } from '@rocket.chat/core-typings';
+import type { Options, Root } from '@rocket.chat/message-parser';
+import type { UseQueryResult } from '@tanstack/react-query';
 import { useMemo } from 'react';
 
-export const useMessageBody = ({ attachments, msg }: Partial<Pick<IMessage, 'msg' | 'attachments'>> = {}): string =>
-	useMemo(() => {
-		if (msg) {
-			return msg;
+import { parseMessageTextToAstMarkdown } from '../lib/parseMessageTextToAstMarkdown';
+import { useAutoTranslate } from './useAutoTranslate';
+
+export const useMessageBody = (
+	parentMessage: UseQueryResult<IMessage, unknown>,
```

Comment:
`nitpick` I don't think we need to couple the data fetching logic in this hook. I would type `parentMessage` as `IMessage | undefined` and let the hook caller handle the loading and error states.